dev-libs/c_blosc/patches/c_blosc-1.13.7.patchset

```diff
@@ -0,0 +1,59 @@
+From d014d1b144f867dbfc158eea1f45d26f9a35bf58 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Zolt=C3=A1n=20Mizsei?= <zmizsei@extrowerk.com>
+Date: Thu, 7 Dec 2017 21:27:35 +0100
+Subject: [PATCH] Build fix
+
+---
+ bench/CMakeLists.txt | 4 ++--
+ bench/bench.c        | 2 +-
+ tests/test_common.h  | 2 +-
+ 3 files changed, 4 insertions(+), 4 deletions(-)
+
+diff --git a/bench/CMakeLists.txt b/bench/CMakeLists.txt
+index 14e74de..1aad1f6 100644
+--- a/bench/CMakeLists.txt
++++ b/bench/CMakeLists.txt
+@@ -4,12 +4,12 @@ set(SOURCES bench.c)
+ 
+ # targets
+ add_executable(bench ${SOURCES})
+-if(UNIX AND NOT APPLE)
++if(UNIX AND NOT APPLE AND NOT HAIKU)
+   # cmake is complaining about LINK_PRIVATE in original PR
+   # and removing it does not seem to hurt, so be it.
+   # target_link_libraries(bench LINK_PRIVATE rt)
+   target_link_libraries(bench rt)
+-endif(UNIX AND NOT APPLE)
++endif(UNIX AND NOT APPLE AND NOT HAIKU)
+ target_link_libraries(bench blosc_shared)
+ 
+ # have to copy blosc dlls on Windows
+diff --git a/bench/bench.c b/bench/bench.c
+index 9852313..4c6196a 100644
+--- a/bench/bench.c
++++ b/bench/bench.c
+@@ -36,7 +36,7 @@
+   #include <mach/mach.h>
+   #include <time.h>
+   #include <sys/time.h>
+-#elif defined(__unix__)
++#elif defined(__unix__) || defined(__HAIKU__)
+   #include <unistd.h>
+   #if defined(__GLIBC__)
+     #include <time.h>
+diff --git a/tests/test_common.h b/tests/test_common.h
+index 4e40533..7ed56f7 100644
+--- a/tests/test_common.h
++++ b/tests/test_common.h
+@@ -68,7 +68,7 @@ static void* blosc_test_malloc(const size_t alignment, const size_t size)
+ #elif defined(_WIN32)
+   /* A (void *) cast needed for avoiding a warning with MINGW :-/ */
+   block = (void *)_aligned_malloc(size, alignment);
+-#elif _POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600
++#elif _POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600 || defined(__HAIKU__)
```

Comment:
`_POSIX_C_SOURCE` should be defined by the application itself. You should look around to see if any header used by the program does so. If not, you should patch it as to conform the [POSIX specification][0].

> #### The _POSIX_C_SOURCE Feature Test Macro
> 
> A POSIX-conforming application should ensure that the feature test macro _POSIX_C_SOURCE is defined before inclusion of any header.
> 
> When an application includes a header described by IEEE Std 1003.1-2001, and when this feature test macro is defined to have the value 200112L:
> 
> 1. All symbols required by IEEE Std 1003.1-2001 to appear when the header is included shall be made visible.
> 
> 2. Symbols that are explicitly permitted, but not required, by IEEE Std 1003.1-2001 to appear in that header (including those in reserved name spaces) may be made visible.
> 
> 3. Additional symbols not required or explicitly permitted by IEEE Std 1003.1-2001 to be in that header shall not be made visible, except when enabled by another feature test macro.
> 
> Identifiers in IEEE Std 1003.1-2001 may only be undefined using the **#undef** directive as described in Use and Implementation of Functions or The Name Space. These **#undef** directives shall follow all **#include** directives of any header in IEEE Std 1003.1-2001.

[0]: http://pubs.opengroup.org/onlinepubs/007904975/functions/xsh_chap02_02.html

Reply:
@alaviss, would you say that a correct upstream patch would be to insert this bloc:
```
#ifdef __HAIKU__
#define _POSIX_C_SOURCE 200112L
#endif /* defined(__HAIKU__) */
```
in a header that is included before the inclusion of `test_common.h` ?
BTW, would you say that each compilation unit should see this bloc, even if does not include `test_common.h`?

Reply:
Found this CMake [snippet][0] that would be interesting as a patch. And yes, this is an upstream issue that should be fixed.

[0]: https://github.com/kcat/openal-soft/commit/4e6acfa0d549281efcbe8293d95b76a37deb370d#diff-af3b638bc2a3e6c650974192a53c7291R102